src/App.jsx

```diff
@@ -4,14 +4,29 @@ import Header from './components/Header/Header';
 import Login from './components/Login/Login';
 import Registration from './components/Registration/Registration';
 import CourseInfo from './components/CourseInfo/CourseInfo';
-import CreateCourse from './components/CreateCourse/CreateCourse';
+import CourseForm from './components/CourseForm/CourseForm';
 import Courses from './components/Courses/Courses';
 import AuthenticatedRoute from './components/Routes/AuthenticatedRoute';
 import UnauthenticatedRoute from './components/Routes/UnauthenticatedRoute';
+import PrivateRoute from './components/PrivateRouter/PrivateRouter';
 
 import './App.css';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { getCurrentUser } from './services';
+import { setUserRole } from './store/user/actionCreators';
 
 function App() {
+	const dispatch = useDispatch();
+	const token = localStorage.getItem('token');
+	useEffect(() => {
+		if (token) {
+			getCurrentUser(token).then((response) =>
+				dispatch(setUserRole(response.data.result.role))
+			);
+		}
+	}, [dispatch, token]);
```

Comment:
Every time something happens with the DOM (e.g. add new author), you get user info which is not correct, this information should be requested only on pages load. Please check network tab when you do some actons

Reply:
Fixed!

Reply:
Not exactly, you shouldn't fetch info in the App.jsx. you should fetch courses and authors on Courses page, and in Course info page the info about the course should be fetched from 
![image](https://user-images.githubusercontent.com/28926420/155726334-8414ff95-3290-475e-b168-d22714bfb891.png)
![image](https://user-images.githubusercontent.com/28926420/155726413-73068f6a-b809-49fd-9135-e2a01d7626e3.png)